### PR TITLE
Use Tag along with Id to identify the notification

### PR DESCRIPTION
**Pull request by iontech**
The pair `(tag, id)` uniquely identify a notification.

Refer [The NotificationManager class description](http://developer.android.com/reference/android/app/NotificationManager.html)

### CODE REVIEW

**Comment by iontech** [comment]
I signed it

File: library/src/main/java/com/bumptech/glide/request/target/NotificationTarget.java [hunk 1]
```diff
@@ -29,35 +30,58 @@
...
+   * @param notificationId  The notificationId of the Notification that we want to load the Bitmap.
+   * @param notificationTag The notificationTag of the Notification that we want to load the Bitmap.
+   *                        May be {@code null}.
    */
-  public NotificationTarget(Context context, int width, int height,
-      int viewId, RemoteViews remoteViews, Notification notification, int notificationId) {
+  public NotificationTarget(Context context, int viewId, RemoteViews remoteViews,
+      Notification notification, int notificationId, String notificationTag) {
+    this(context, SIZE_ORIGINAL, SIZE_ORIGINAL, viewId, remoteViews, notification, notificationId,
+      notificationTag);
```
Comment (TWiStErRob) [inline library/src/main/java/com/bumptech/glide/request/target/NotificationTarget.java]: I'm a little torn here. The `notify` and `cancel` methods in the Android API break tradition of putting the optional args last, so the order becomes inconsistent with these constructors and even the doc says `(tag, id)`.

I guess we can leave it as is to help understanding the telescoping constructors better. This and the widget target is probably in need of a builder pattern (out of scope for you now).

**Comment by TWiStErRob** [comment]
One comment on changes above and please squash your commits and push to your branch, other than that lgtm.

**Comment by sjudd** [comment]
LGTM. @TWiStErRob I agree we could use a builder here and that it's out of scope. I think it's fine to leave the constructor as id, tag, though I share your feelings :).

**Comment by TWiStErRob** [comment]
@iontech please do a squash on your branch so it has only one commit.

**Comment by iontech** [comment]
@TWiStErRob I apologize, it took awhile. Pushed the squashed commit. Should be good.

**Pull request merged**